packages/grafana-ui/src/components/Combobox/Combobox.tsx

```diff
@@ -17,15 +18,33 @@ export type ComboboxOption<T extends string | number = string> = {
   description?: string;
 };
 
-interface ComboboxProps<T extends string | number>
-  extends Omit<InputProps, 'prefix' | 'suffix' | 'value' | 'addonBefore' | 'addonAfter' | 'onChange'> {
+interface ComboboxBaseProps<T extends string | number>
+  extends Omit<InputProps, 'prefix' | 'suffix' | 'value' | 'addonBefore' | 'addonAfter' | 'onChange' | 'width'> {
   isClearable?: boolean;
   createCustomValue?: boolean;
   options: Array<ComboboxOption<T>>;
   onChange: (option: ComboboxOption<T> | null) => void;
   value: T | null;
+  /**
+   * Defaults to 100%. Number is a multiple of 8px. 'auto' will size the input to the content.
+   * */
+  width?: number | 'auto';
 }
 
+type AutoSizeConditionals =
+  | {
+      width: 'auto';
+      minWidth: number;
+      maxWidth?: number;
+    }
+  | {
+      width?: number;
+      minWidth?: never;
+      maxWidth?: never;
```

Comment:
@joshhunt Should we keep these when we have a defined width, and only change the optionality rather than disallowing the value?

Reply:
No, i don't think so. I think this is good 👍